06-classification_continued.Rmd

```diff
@@ -362,21 +451,24 @@ than 5-fold:
 ```{r 06-10-fold}
 cancer_vfold <- vfold_cv(cancer_train, v = 10, strata = Class)
 
-workflow() %>%
-  add_recipe(cancer_recipe) %>%
-  add_model(knn_spec) %>%
-  fit_resamples(resamples = cancer_vfold) %>%
-  collect_metrics()
+vfold_metrics <- workflow() %>%
+                  add_recipe(cancer_recipe) %>%
+                  add_model(knn_spec) %>%
+                  fit_resamples(resamples = cancer_vfold) %>%
+                  collect_metrics()
+vfold_metrics
 ```
 
 ### Parameter value selection
 
 Using 5- and 10-fold cross-validation, we have estimated that the prediction
-accuracy of our classifier is somewhere around 88%. Whether 88% is good or not
+accuracy of our classifier is somewhere around `r round(100*(vfold_metrics %>% filter(.metric == "accuracy"))$mean,0)`%. 
+Whether that is good or not
 depends entirely on the downstream application of the data analysis. In the
 present situation, we are trying to predict a tumour diagnosis, with expensive,
 damaging chemo/radiation therapy or patient death as potential consequences of
-misprediction. Hence, we'd like to do better than 88% for this application.  
+misprediction. Hence, we might like to 
+do better than `r round(100*(vfold_metrics %>% filter(.metric == "accuracy"))$mean,0)`% for this application.  
 
 In order to improve our classifier, we have one choice of parameter: the number of
 neighbours, $K$. Since cross-validation helps us evaluate the accuracy of our
```

Comment:
Students were really confused about how `grid = 10` chooses the 10 values. It is not well enough explained in the textbook here, and the documentation is NOT accessible:

"If no tuning grid is provided, a semi-random grid (via `dials::grid_latin_hypercube()`) is created with 10 candidate parameter combinations."

So if we don't want to explain this in the text here, we could avoid that by just always giving a data frame of tuning combinations (which we do next in the book and is MUCH easier to explain).

Reply:
done